lib/Doctrine/ORM/Persisters/ManyToManyPersister.php

```diff
@@ -37,130 +37,288 @@ class ManyToManyPersister extends AbstractCollectionPersister
 {
     /**
      * {@inheritdoc}
-     *
-     * @override
      */
-    protected function getDeleteRowSQL(PersistentCollection $coll)
+    public function delete(PersistentCollection $coll)
     {
-        $columns    = array();
-        $mapping    = $coll->getMapping();
-        $class      = $this->em->getClassMetadata(get_class($coll->getOwner()));
-        $tableName  = $this->quoteStrategy->getJoinTableName($mapping, $class, $this->platform);
+        $mapping = $coll->getMapping();
 
-        foreach ($mapping['joinTable']['joinColumns'] as $joinColumn) {
-            $columns[] = $this->quoteStrategy->getJoinColumnName($joinColumn, $class, $this->platform);
+        if ( ! $mapping['isOwningSide']) {
+            return; // ignore inverse side
         }
 
-        foreach ($mapping['joinTable']['inverseJoinColumns'] as $joinColumn) {
-            $columns[] = $this->quoteStrategy->getJoinColumnName($joinColumn, $class, $this->platform);
+        $this->conn->executeUpdate($this->getDeleteSQL($coll), $this->getDeleteSQLParameters($coll));
```

Comment:
The real advantage is using DBAL properly instead of using it in a broken way: ``executeQuery`` is meant to be used for read queries only. When using the MasterSlaveConnection, ``executeQuery`` can be executed against the slave, while ``executeUpdate`` forces switching to the master